crates/macro/src/parser/mod.rs

```diff
@@ -28,14 +29,23 @@ use crate::parser::attrs::{check_recorded_struct_for_impl, record_struct};
 
 static GENERATOR_STRUCT: OnceLock<Mutex<HashMap<String, bool>>> = OnceLock::new();
 
-static REGISTER_INDEX: AtomicUsize = AtomicUsize::new(0);
+static REGISTER_INDICES: OnceLock<DashMap<String, usize>> = OnceLock::new();
 
 fn get_register_ident(name: &str) -> Ident {
-  let new_name = format!(
-    "__napi_register__{}_{}",
-    rm_raw_prefix(name),
-    REGISTER_INDEX.fetch_add(1, std::sync::atomic::Ordering::Relaxed)
-  );
+  let register_indices = REGISTER_INDICES.get_or_init(DashMap::new);
+  let index = match register_indices.entry(name.to_string()) {
+    dashmap::mapref::entry::Entry::Occupied(mut occupied) => {
+      let value = occupied.get_mut();
+      let return_value = *value;
+      *value += 1;
+      return_value
+    }
+    dashmap::mapref::entry::Entry::Vacant(vacant) => {
+      vacant.insert(1);
+      0
+    }
+  };
+  let new_name = format!("__napi_register__{}_{}", rm_raw_prefix(name), index,);
```

Comment:
I implemented this so that the register name is unique. But I don't know if this is needed.
Even if the register name is deconflicted here, I still get a duplicate type and an export.
```ts
// wasi.cjs
module.exports.FOO = __napiModule.exports.FOO
module.exports.FOO = __napiModule.exports.FOO

// index.cjs
module.exports.FOO = nativeBinding.FOO
module.exports.FOO = nativeBinding.FOO

// index.d.ts
export const FOO: number
export const FOO: number
```
Does it make sense to return an error if it's already used instead?

Reply:
I kind of forgot the context back then... Let me recall first, and I'll reply to you once I remember.